lib/core/api.js

```diff
@@ -359,8 +359,13 @@ function addDataToDataset(data, datasetId, blockIndex, uploadType, callback) {
       callback('duplicate', jqxhr.responseJSON.index);
     } else {
       api.log('platform add data to dataset failed.');
-      api.log(status);
-      callback(status);
+      var err = new Error('Uploading data to platform failed.');
+      err.name = 'API Error';
+      err.status = status;
+      err.datasetId = datasetId;
+      err.sessionId = status.meta.trace.request;
```

Comment:
The `status.meta.trace.request` is the trace identifier used *per request* generated by the server, which is definitely a good thing to show in the error. However, I would save it under `err.requestId`.

The session id was talking about was something sent by the client (the uploader) with one or more requests. We don't need to add it now, but would be good to add in the future.

For more info, look for "X-Tidepool-Trace-Session" in https://docs.google.com/a/tidepool.org/document/d/1RU6-4HXCq4FRnh186OSNSjEl0jDYcN4JSf-bmDh0Sn0/edit?usp=sharing.